mods/ra/maps/soviet-08a/soviet08a.lua

```diff
@@ -0,0 +1,161 @@
+		Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+		scripteddrop.SendParatroopers(waypoint35.CenterPosition, false, 10)
+	end)
+end
+
+ChurchAmbushTrigger = function()
+	if not AmbushSwitch then
+		local hiding = Reinforcements.Reinforce(germany, { 'e1', 'e1', 'e1', 'e1', 'e1', 'e1', 'e1', 'e3', 'e3', 'e3' }, { waypoint78.Location, waypoint5.Location })
+		Utils.Do(hiding, function(actor)
+			IdleHunt(actor)
+		end)
+	end
+	AmbushSwitch = true
+end
+
+Obj2TriggerFunction = function()
+	ussr.MarkCompletedObjective(DestroyVillageObjective)
+	Media.PlaySpeechNotification(ussr, "ReinforcementsArrived")
+	local reinforcement = ussrReinforcements.mammoth
+	Reinforcements.ReinforceWithTransport(ussr, "lst.reinforcement", reinforcement.actors, reinforcement.entryPath, reinforcement.exitPath)
+end
+
+AddReinforcmentTriggers = function()
+	AddEastReinforcementTrigger()
+	AddSouthReinforcementTrigger()
+	AddParadropReinforcementTrigger()
+end
+
+AddRetreatTrigger = function()
+	Trigger.OnEnteredProximityTrigger(Actor222.CenterPosition, WDist.FromCells(12), function(actor, id)
+		if actor.Owner == ussr and actor.Type == "barr" then
+			alliedScouts = Utils.Where(alliedScouts, function(scout) return not scout.IsDead end)
+			local removed
+			Utils.Do(alliedScouts, function(scout)
+				if scout.Type == "e1" and not removed then
+					removed = true
+				else
+					scout.Stop()
+					scout.Move(waypoint1.Location, 1)
+				end
+			end)
+			Trigger.RemoveProximityTrigger(id)
+		end
+	end)
+end
+
+Tick = function()
+	greece.Cash = 1000
+
+	if greece.HasNoRequiredUnits() and germany.HasNoRequiredUnits() then
+		ussr.MarkCompletedObjective(KillAll)
+	end
+
+	if ussr.HasNoRequiredUnits() then
+		greece.MarkCompletedObjective(BeatUSSR)
+	end
+end
+
+WorldLoaded = function()
+	ussr = Player.GetPlayer("USSR")
+	germany = Player.GetPlayer("Germany")
+	greece = Player.GetPlayer("Greece")
+	
+	Trigger.OnObjectiveAdded(ussr, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "New " .. string.lower(p.GetObjectiveType(id)) .. " objective")
+	end)
+	
+	KillAll = ussr.AddPrimaryObjective("Destroy all Allied units and structures.")
+	DestroyVillageObjective = ussr.AddSecondaryObjective("Destroy the village of Allied sympathizers.")
+	BeatUSSR = greece.AddPrimaryObjective("Defeat the Soviet forces.")
+	
+	Trigger.OnObjectiveCompleted(ussr, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective completed")
+	end)
+	Trigger.OnObjectiveFailed(ussr, function(p, id)
+		Media.DisplayMessage(p.GetObjectiveDescription(id), "Objective failed")
+	end)
+
+	Trigger.OnPlayerLost(ussr, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(ussr, "MissionFailed")
+		end)
+	end)
+	Trigger.OnPlayerWon(ussr, function()
+		Trigger.AfterDelay(DateTime.Seconds(1), function()
+			Media.PlaySpeechNotification(ussr, "MissionAccomplished")
+		end)
+	end)
+
+	Harvester.FindResources()
```

Comment:
This loos odd. Doesn't the harvester start mining on its own?

Reply:
The refinery spawned one will. This is to get the one at 57,56 going.